src/environment.rs

```diff
@@ -33,6 +33,51 @@ use std::{
     time::Duration,
 };
 
+/// Verify the location of the prefix folder is not changed so the applied prefix path is still valid.
+/// Errors when there is a file system error or the path does not align with the defined prefix.
+/// Returns false when the file is not present.
+pub fn verify_prefix_location_unchanged(prefix_file: &Path) -> miette::Result<()> {
+    match std::fs::read_to_string(prefix_file) {
+        // Not found is fine as it can be new or backwards compatible.
+        Err(e) if e.kind() == ErrorKind::NotFound => Ok(()),
+        // Scream the error if we dont know it.
+        Err(e) => Err(e).into_diagnostic(),
+        // Check if the path in the file aligns with the current path.
+        Ok(p) => {
+            if prefix_file.starts_with(&p) {
+                Ok(())
+            } else {
+                Err(miette::miette!(
+                    "The project location seems to be change from `{}` to `{}`, this is not allowed.\
+                \nPlease remove the `{}` folder and run again",
+                    p,
+                    prefix_file
+                        .parent()
+                        .expect("prefix_file should always be a file")
+                        .display(),
+                    consts::PIXI_DIR
+                ))
+            }
+        }
+    }
+}
+
+/// Create the prefix location file.
+/// Give it the file path of the required location to place it.
+fn create_prefix_location_file(prefix_file: &Path) -> miette::Result<()> {
+    let parent = prefix_file
+        .parent()
+        .ok_or_else(|| miette::miette!("Cannot find parent of '{}'", prefix_file.display()))?;
+
+    if parent.exists() {
+        let contents = parent.to_str().ok_or_else(|| {
+            miette::miette!("Failed to convert path to str: '{}'", parent.display())
```

Comment:
It would be good to add more context here. What does this path refer to? Same goes for the error above. You could also add more context where you invoke this function. Using the `with_context` method.

Reply:
`with_context` :open_mouth:  What an awesome library!